### DISCUSSION TITLE
Getting an undefined method error calling ConnectAPI::App::all_build_train_numbers

### DISCUSSION THREAD

**Discussion by markpar** [fastlane/fastlane#17294] [Q&A] [answered]
Hi!  I'm using Fastlane 2.160.0 and am wanting to use Spaceship to add a test group to my app's TestFlight build.  I followed the directions in https://github.com/fastlane/fastlane/blob/master/spaceship/docs/AppStoreConnect.md and can successfully log in and find my app, which I verified by printing the value of `app.bundle_id`.  However, when I call `app.all_build_train_numbers`, I am getting `[!] undefined method 'all_build_train_numbers' for #<Spaceship::ConnectAPI::App:0x007f822913f378> (NoMethodError)`.

I've tried this in straight Ruby as well as in a Fastfile and get the same error.  Any troubleshooting tips would be appreciated!  Please let me know if there is any additional information I can provide.  🙂 

```ruby
  token = Spaceship::ConnectAPI::Token.create(
    key_id: 'OUR_ID',
    issuer_id: 'OUR_ISSUER',
    filepath:  File.absolute_path("OUR_PRIVATE_KEY")
  )

  Spaceship::ConnectAPI.token = token
  app = Spaceship::ConnectAPI::App.find("OUR_BUNDLE_ID")
  puts app.bundle_id  # Succeeds
  puts app.all_build_train_numbers  # Fails with NoMethodError :(
```

The equivalent `Spaceship::Tunes` code does work for me.
```ruby
  Spaceship::Tunes.login("USER", "PW")
  app2 = Spaceship::Tunes::Application.find("OUR_BUNDLE_ID")
  puts app2.bundle_id  # Succeeds
  puts app2.all_build_train_numbers # Succeeds
```

**Comment by max-ott** [comment, marked as answer]
`all_build_train_numbers` is a legacy thing based on `Tunes`. You cannot simply mix and match `Spaceship::ConnectAPI` with `Spaceship::Tunes`

**Comment by markpar** [comment]
Thanks for the reply, @max-ott!  I was following the instructions on the doc page I linked above and it wasn't clear to me that the example code only worked with `Tunes`.

![image](https://user-images.githubusercontent.com/6062009/93904878-2a74de80-fcaf-11ea-9207-881d8e80110f.png)

After reading through the source more, it looks like `app.get_builds` followed by `app.add_beta_groups_to_build` is what I'm looking for.
